spec/models/season/phase_switcher_spec.rb

```diff
@@ -4,32 +4,61 @@
 context 'when switching phases' do
   subject { Season.current }
 
-  describe '#fake_application_phase' do
-    it 'timeshifts the application phase to today' do
-      Timecop.travel(Season.current.applications_close_at - 1.month) do
-        Season::PhaseSwitcher.fake_application_phase
-        expect(subject).to be_application_period
+  describe 'destined' do
+
+    it 'sends the params to a phase setting method' do
+      phase = 'fake_application_phase'
+      Season::PhaseSwitcher.destined(phase)
+      expect(subject).to_not be_nil
+    end
+
+    context 'when directed to the #fake_application_phase' do
+      it 'sets date range so that the season is open for application submissions' do
+        phase = 'fake_application_phase'
+        Timecop.travel(Season.current.applications_close_at - 1.month) do
+          Season::PhaseSwitcher.destined(phase)
+          expect(subject).to_not be_nil
+          expect(subject).to be_application_period
+        end
       end
     end
-  end
 
-  describe '#fake_coding_phase' do
-    it 'timeshifts the coding phase to today' do
-      Timecop.travel(Season.current.ends_at - 1.week) do
-        Season::PhaseSwitcher.fake_coding_phase
-        expect(subject).to be_started
+    context 'when directed to the #fake_coding_phase' do
+      it 'sets dates so that the summer of code is currently happening' do
+        phase = 'fake_coding_phase'
+        Timecop.travel(Season.current.ends_at - 1.week) do
+          Season::PhaseSwitcher.destined(phase)
+          expect(subject).to be_started
+        end
       end
     end
-  end
 
-  describe '#fake_proposals_phase' do
-    it 'timeshifts the proposal period to today' do
-      Timecop.travel(Season.current.project_proposals_close_at - 2.weeks) do
-        fake_time = Time.now #as returned by Timecop
-        Season::PhaseSwitcher.fake_proposals_phase
-        expect(subject.project_proposals_open_at).to be < fake_time
-        expect(subject.project_proposals_close_at).to be > fake_time
+    context 'when directed to the #fake_proposals_phase' do
+      it 'set dates so that season is open for project proposals' do
+        phase = 'fake_proposals_phase'
+        Timecop.travel(Season.current.project_proposals_close_at - 2.weeks) do
+          fake_time = Time.now #as returned by Timecop
+          Season::PhaseSwitcher.destined(phase)
+          expect(subject.project_proposals_open_at).to be < fake_time
+          expect(subject.project_proposals_close_at).to be > fake_time
+        end
       end
     end
+
+    it 'addresses the requested phase' do
+      another_phase = 'fake_proposals_phase'
+
+      Season::PhaseSwitcher.destined(another_phase)
+      # Just checking, here too the error is not raised
+      Season.current
+      allow_any_instance_of(Season).to receive(:save).and_raise RuntimeError
+      expect(subject).to_not be_application_period
+    end
+
+    it 'fails silently when it receives a non-whitelisted phase' do
```

Comment:
If you go with the `ArgumentError` approach, this needs to be updated as well of course 😉